### PR TITLE
Fix the loss initialization in intermediate_source/char_rnn_generation_tutorial.py

### LINKED ISSUE #844: Typo instantiating loss at zero
https://github.com/pytorch/tutorials/blob/48e5cccecbd385e00b95cb60ecdfecd06a43282b/intermediate_source/char_rnn_generation_tutorial.py#L281

Hello folks at Pytorch,

When instantiating loss at a real zero, i think you rather meant something like (i don't know if there is a better way but I usually do it this way) : 

`loss = torch.autograd.Variable(torch.Tensor([0]))`

### PR DESCRIPTION
**Pull request by akjalok**
Fixes #844

## Description
Changed the loss initialization from 0 to torch.Tensor([0]) in order to avoid confusion about the dtypes.

## Checklist

- [x] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x]  Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnessessary issues are included into this pull request.
cc @svekars @carljparker @kit1980

### CODE REVIEW

File: intermediate_source/char_rnn_generation_tutorial.py [hunk 1]
```diff
@@ -278,7 +278,7 @@ def train(category_tensor, input_line_tensor, target_line_tensor):
 
     rnn.zero_grad()
 
-    loss = 0
+    loss = torch.Tensor([0])
```
Comment (NicolasHug) [inline intermediate_source/char_rnn_generation_tutorial.py:281]: This works but it should be

```suggestion
    loss = torch.tensor([0])
```

**Review by NicolasHug — APPROVED** [review]
I'll approve, but as mentioned in the original issue https://github.com/pytorch/tutorials/issues/844#issuecomment-1571271807 the current code is just fine. We don't need to initialize `loss` as a tensor.

File: intermediate_source/char_rnn_generation_tutorial.py [hunk 2]
```diff
@@ -278,7 +278,7 @@ def train(category_tensor, input_line_tensor, target_line_tensor):
 
     rnn.zero_grad()
 
-    loss = 0
+    loss = torch.tensor([0]) # in PyTorch 2.0 and later can be don't need to initialize as tensor, can be ``loss = 0``
```
Comment (NicolasHug) [inline intermediate_source/char_rnn_generation_tutorial.py:281]: I believe this is true for any pytorch version, it's not related to 2.0

Comment (svekars) [inline intermediate_source/char_rnn_generation_tutorial.py:281]:
```suggestion
    loss = torch.tensor([0]) # you can also just simply use ``loss = 0``
```

File: intermediate_source/char_rnn_generation_tutorial.py [hunk 3]
```diff
@@ -278,7 +278,7 @@ def train(category_tensor, input_line_tensor, target_line_tensor):
 
     rnn.zero_grad()
 
-    loss = 0
+    loss = torch.tensor([0]) # you can also just simply use ``loss = 0``
```
Comment (NicolasHug) [inline intermediate_source/char_rnn_generation_tutorial.py:281]:
```suggestion
    loss = torch.Tensor([0]) # you can also just simply use ``loss = 0``
```
